docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -19,71 +19,28 @@ Each project stores blocks in different places and uses slightly different block
 
 For configuration of remote write to Grafana Mimir, refer to [Configuring Prometheus remote write]({{< relref "../operators-guide/securing/authentication-and-authorization.md#configuring-prometheus-remote-write" >}}).
 
-## Uploading historic blocks to the Grafana Mimir storage bucket
+## Uploading historic TSDB blocks to Grafana Mimir
+
+Grafana Mimir supports uploading of historic TSDB blocks, also from other systems such as Prometheus and Thanos.
+In order to enable this functionality, either for all tenants or a specific one, refer to
+[Configuring TSDB block upload]({{< relref "../operators-guide/configuring/configuring-tsdb-block-upload.md" >}}).
 
 Prometheus stores TSDB blocks in the path specified in the `--storage.tsdb.path` flag.
 
-To find all blocks directories in the TSDB `<STORAGE TSDB PATH>`, run the following command:
+To find all block directories in the TSDB `<STORAGE TSDB PATH>`, run the following command:
 
 ```bash
 find <STORAGE TSDB PATH> -name chunks -exec dirname {} \;
 ```
 
-Grafana Mimir supports multiple tenants and stores blocks with a tenant prefix.
-With multi-tenancy disabled, there is a single tenant called `anonymous`.
-
-Copy each directory output from the previous command to the Grafana Mimir object storage bucket with
-your tenant prefix.
+Grafana Mimir supports multiple tenants and stores blocks per tenant. With multi-tenancy disabled, there
+is a single tenant called `anonymous`.
 
-- For AWS S3, use the `aws` tool in the following command:
+Use Grafana mimirtool to upload each block, e.g. identified by the previous command, to Grafana Mimir:
 
 ```bash
-aws s3 cp <DIRECTORY> s3://<TENANT>/<DIRECTORY>
+mimirtool backfill --user=<tenant> --key=$(cat token.txt) --address=http://<mimir-hostname> --id=<tenant> <block1> <block2>...
 ```
 
-- For Google Cloud Storage (GCS), use the `gsutil` tool in the following command:
-
-```bash
-gsutil -m cp -r <DIRECTORY> gs://<TENANT>/<DIRECTORY>
-```
-
-## Block metadata
```

Comment:
@pracucci sorry, I just resolved a merge conflict heavy-handedly since I'm trying to concentrate on mimirtool work at the moment. I'm going to fix it up before moving the PR out of the WIP stage :)

Reply:
I added it back.